---
title: "Microbiome"
author: "Abdulmalik_Oladipupo"
date: "2025-04-21"
output: html_document
---

### Install and Load all required packages
```{r}
# Check and install missing packages only
required_packages <- c(
    "vegan", "phyloseq", "Biostrings", "metagenomeSeq", 
    "decontam", "indicspecies", "DESeq2", "ggpubr", "tidyverse", 
    "ggplot2", "dunn.test", "writexl", "pairwiseAdonis", "remotes", 
    "cluster", "devtools", "statmod", "gtools", "BiocStyle", "DECIPHER", "phangorn", "gridExtra"
)

# Install missing packages
for (pkg in required_packages) {
    if (!requireNamespace(pkg, quietly = TRUE)) {
        cat("Installing missing package:", pkg, "\n")
        if (pkg %in% c("vegan", "phyloseq", "Biostrings", "metagenomeSeq", 
                       "decontam", "indicspecies", "DESeq2", "ggpubr", "BiocStyle", "DECIPHER", "phangorn")) {
            BiocManager::install(pkg, lib = "~/R/library")
        } else {
            install.packages(pkg, repos = "http://cran.us.r-project.org", lib = "~/R/library")
        }
    }
}

# Load libraries
library("vegan")
library("phyloseq")
library("Biostrings")
library("metagenomeSeq")
library("decontam")
library("indicspecies")
library("DESeq2")
library("ggpubr")
library("tidyverse")
library("ggplot2")
library("dunn.test")
library("writexl")
library("pairwiseAdonis")
library("cluster")
library("devtools")
```

### Set up color pallet to use through this analysis
```{r setup, include = FALSE}
# color blind pallet
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
```

### Load in all required formatted files from the HPC or formatted epi2me output. 

```{r setup, include = FALSE}
taxa <- read.csv("R_data/Taxa_file.csv", header = TRUE, row.names = 1, na.strings = "na")
otu.table <- read.csv("R_data/OTU_table_16s.csv", na.strings = "na")
metadata <- read.csv("R_data/metadata_test.csv", na.strings = "na")
tree.read <- "R_data/tree.nwk"
fasta.read <- "R_data/fasta_file.fasta"
```

### Format all CSV input data to ensure match rows and column
```{r}
taxa$OTU <- rownames(taxa)
rownames(otu.table) <- otu.table$OTU
otu.table <- otu.table[,-1]
rownames(metadata) <- metadata$SampleID #row names must match OTU table headers
```


### Load all file into phyloseq
```{r}
phy.tax <- phyloseq::tax_table(as.matrix(taxa))
phy.OTU.table <- phyloseq::otu_table(otu.table, taxa_are_rows = TRUE)
phy.data <- phyloseq::sample_data(metadata)
phy.fasta <- Biostrings::readDNAStringSet(fasta.read, format="fasta", seek.first.rec=TRUE, use.names=TRUE)
phy.tree <- phyloseq::read_tree(tree.read)

phyloseq.data <- phyloseq(phy.tax, phy.OTU.table, phy.data, phy.fasta, phy.tree)

```
### Samples quality control and read distribution

```{r}
# filter out taxa that have non-zero abundance in less than 2 samples
ps.clean <- filter_taxa(phyloseq.data, function (x) {sum(x > 0) >= 2}, prune=TRUE)
# see read distribution from the highest to lowest
sort(sample_sums(ps.clean), decreasing = T) # read distribution
# check total reads
sum(sample_sums(ps.clean))
# chech mean read 
mean(sample_sums(ps.clean))
# median read
median(sample_sums(ps.clean))
```

### Analysis and Visualization

#### ALPHA DIVERSITY

```{r}
# Estimate multiple richness measures at once
alpha_diversity <- estimate_richness(ps.clean, measures = c("Shannon", "InvSimpson", "Observed"))

# Add the measures to the sample data
ps.clean@sam_data$shannon <- alpha_diversity$Shannon
ps.clean@sam_data$invsimpson <- alpha_diversity$InvSimpson
ps.clean@sam_data$richness <- alpha_diversity$Observed

# Compute evenness (Shannon / log(richness))
ps.clean@sam_data$even <- ps.clean@sam_data$shannon / log(ps.clean@sam_data$richness)
```

```{r}
# extract data from phyloseq object into a dataframe
sample.df <- data.frame(ps.clean@sam_data)
# arrange order
sample.df$treatment <- factor(sample.df$treatment, levels = c("BASAL", "YCW", "IFC4", "IFC4+YCW"))
```







```{r}
# Shannon diversity by time*treatment interaction
shan.treatment.time <- ggplot(sample.data.bac, aes(x = treatment, y = shannon, fill = treatment)) + 
  geom_boxplot() +
  #geom_jitter() + 
  ylab("Shannon") + 
  stat_compare_means(method = "") + 
  xlab("")+
  theme_bw() +
  facet_wrap(~day) +
  theme(axis.text.x = element_text(size = 11, vjust = 0.5, hjust=0.5, face = "bold"),
        axis.text.y = element_text(size = 12),
        axis.title.y = element_text(size = 14, face = "bold"),
        axis.line = element_line(size = 0.6),
        legend.text = element_text(size = 12),    # Adjust legend text size
        legend.title = element_text(size = 16))     # Adjust legend title size
shan.treatment.time

```

```{r}
shan_trt_time <- aov(shannon ~ treatment*day, data = sample.data.bac)
anova_table <- anova(shan_trt_time) 
print(anova_table) #treatment=0.076, day=0.391, inter=0.928
tukey_results <- TukeyHSD(shan_trt_time)
print(tukey_results)
```


```{r}
# fecal samples relative amount/treatment*time
top20 <- names(sort(taxa_sums(physeq.no.chloro), decreasing=TRUE))[1:12]
ps.top20 <- transform_sample_counts(physeq.no.chloro, function(OTU) OTU/sum(OTU))
ps.top20 <- prune_taxa(top20, ps.top20)
dat.dataframe <- psmelt(ps.top20)
dat.agr <- aggregate(Abundance ~ treatment*day + Species, data = dat.dataframe, FUN = mean)
rel_abun_treatment <-ggplot(dat.agr, aes(x = treatment, y = Abundance, fill = Phylum)) +
  facet_wrap(~day) +
  geom_bar(stat = "identity", position = "fill") +
  theme_classic() +
  labs(title = "",
       x = "Treatments",
       y = "Relative Abundance") +
  theme(axis.text.x = element_text(size = 12, face = "bold", vjust = 0.5, hjust=0.5),
        axis.text.y = element_text(size = 12, face = "bold"),
        axis.title.y = element_text(size = 15, face = "bold"),
        axis.title.x = element_text(size = 15, face = "bold"),
        legend.text = element_text(size = 12),    # Adjust legend text size
        legend.title = element_text(size = 15))     # Adjust legend title size 

rel_abun_treatment
```





































